test/tools/utils.ts

```diff
@@ -499,3 +504,58 @@ export function topologyWithPlaceholderClient(
     options as TopologyOptions
   );
 }
+
+export async function itInNodeProcess(
+  title: string,
+  fn: (d: { expect: typeof import('chai').expect; mongodb: typeof import('../mongodb') }) => void
+) {
+  it(title, async () => {
+    const script = `
+      import { expect } from 'chai';
+      import * as mongodb from './test/mongodb';
+      const run = ${fn};
+      run({ expect, mongodb }).then(
+        () => process.exit(0),
```

Comment:
```suggestion
        () => process.exitCode = 0,
```


Just a very minor note, but often it's preferable to use `.exitCode` over `.exit(code)` because the latter immediately ends the process and wouldn't e.g. wait for unhandled rejections or for I/O to finish

Reply:
I agree, the original implementation of this launched node with `mjs` and if you `import` process, you get this error:
```txt
TypeError: Cannot assign to read only property 'exitCode' of object '[object Module]'
```

Now it uses `ts-node` and `.cts` just curious if you know of how ^ this should be done in esm?

Reply:
Hmm, what exactly did you try originally here?

```js
$ cat test.mjs
import process from 'process'; // optional line

process.exitCode = 42;
$ node test.mjs; echo $?
42
```

seems to work fine.

It doesn’t work with `import * as process from 'process';`, but then again star-import is probably not what you want to do if you just want normal access to the `process` object, right?

Reply:
I am in the (possibly bad) habit of always using `* as` so that's why it wasn't working, TIL, TY.